src/PrestaShopBundle/Controller/Admin/Improve/International/StateController.php

```diff
@@ -64,4 +76,266 @@ public function getStatesAction(Request $request)
+    /**
+     * Enables states on bulk action
+     *
+     * @AdminSecurity("is_granted('update', request.get('_legacy_controller'))", redirectRoute="admin_states_index")
+     * @DemoRestricted(redirectRoute="admin_states_index")
+     *
+     * @param Request $request
+     *
+     * @return RedirectResponse
+     */
+    public function bulkEnableAction(Request $request): RedirectResponse
+    {
+        $stateIds = $this->getBulkStatesFromRequest($request);
+
+        try {
+            $this->getCommandBus()->handle(new BulkToggleStateStatusCommand(true, $stateIds));
+
+            $this->addFlash(
+                'success',
+                $this->trans('The status has been successfully updated.', 'Admin.Notifications.Success')
+            );
+        } catch (StateException $e) {
+            $this->addFlash('error', $this->getErrorMessageForException($e, $this->getErrorMessages($e)));
+        }
+
+        return $this->redirectToRoute('admin_states_index');
+    }
+
+    /**
+     * Disables states on bulk action
+     *
+     * @AdminSecurity("is_granted('update', request.get('_legacy_controller'))", redirectRoute="admin_states_index")
+     * @DemoRestricted(redirectRoute="admin_states_index")
+     *
+     * @param Request $request
+     *
+     * @return RedirectResponse
+     */
+    public function bulkDisableAction(Request $request): RedirectResponse
+    {
+        $stateIds = $this->getBulkStatesFromRequest($request);
+
+        try {
+            $this->getCommandBus()->handle(new BulkToggleStateStatusCommand(false, $stateIds));
+
+            $this->addFlash(
+                'success',
+                $this->trans('The status has been successfully updated.', 'Admin.Notifications.Success')
+            );
+        } catch (StateException $e) {
+            $this->addFlash('error', $this->getErrorMessageForException($e, $this->getErrorMessages($e)));
+        }
+
+        return $this->redirectToRoute('admin_states_index');
+    }
+
+    /**
+     * @return array
+     */
+    private function getToolbarButtons(): array
+    {
+        $toolbarButtons = [];
+
+        $toolbarButtons['add'] = [
+            'href' => $this->generateUrl('admin_states_create'),
+            'desc' => $this->trans('Add new state', 'Admin.International.Feature'),
+            'icon' => 'add_circle_outline',
+        ];
+
+        return $toolbarButtons;
+    }
+
+    /**
+     * @param Request $request
+     *
+     * @return array
+     */
+    private function getBulkStatesFromRequest(Request $request): array
+    {
+        $stateIds = $request->request->get('state_states_bulk');
+
+        if (!is_array($stateIds)) {
+            return [];
+        }
+
+        foreach ($stateIds as $i => $stateId) {
+            $stateIds[$i] = (int) $stateId;
+        }
+
+        return $stateIds;
+    }
+
+    /**
+     * @param Exception|null $e
+     *
+     * @return array
+     */
+    private function getErrorMessages(Exception $e = null): array
```

Comment:
What's the point of the argument that isn't used in the method ?

Reply:
Understood. Then the argument here is useless for now.

About the bulk delete behaviour, there is no absolute rule. Only a question of functional choice:
1. do nothing if a single error happens
2. do as many as possible, as you suggest
3. do until first error happens, then stop

UX/UI should be taken in consideration in such choice. What is the behavior the end user needs the most ?
Usually the most common choice is all or nothing (option 1) to help the end user who then doesn't have to search for what's left to delete. Flash notification disappears too quickly, and the user has no time to take note.